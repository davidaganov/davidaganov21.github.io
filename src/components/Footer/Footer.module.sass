.footer
  margin-top: auto
  padding-bottom: 3rem

.inner
  display: flex
  flex-direction: column
  align-items: center

.author
  text-align: center
  color: var(--color-white-700) !important
  &:hover
    color: var(--color-white) !important
  @media (min-width: 1000px)
    margin-bottom: .5rem
  @media (min-width: 375px)
    br
      display: none

.social
  @media (max-width: 999px)
    margin: 1rem 0
