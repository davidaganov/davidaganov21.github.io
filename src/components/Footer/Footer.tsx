import cn from "classnames"
import styles from "./Footer.module.sass"
import { Social } from "../"
import { useTranslation } from "react-i18next"

export const Footer = () => {
  const { t } = useTranslation()

  return (
    <footer
      className={styles.footer}
      id="footer"
      role="contentinfo"
    >
      <div className={cn(styles.inner, "inner")}>
        <a
          href="https://github.com/davidaganov21/davidaganov21.github.io"
          className={cn(styles.author, "inline-link inline-link--white")}
          target="_blank"
          rel="noreferrer"
        >
          {t("contacts.description")}&nbsp;
          <br />
          {t("contacts.author")}
        </a>
        <Social className={styles.social} />
        <a
          className="inline-link inline-link--white"
          href="mailto:davidaganov21@gmail.com"
        >
          davidaganov21@gmail.com
        </a>
      </div>
    </footer>
  )
}
