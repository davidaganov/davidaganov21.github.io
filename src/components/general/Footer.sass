.footer
  margin: 3rem 0
  &__inner
    display: flex
    flex-direction: column
    align-items: center
  &__author
    text-align: center
    color: var(--color-white-700) !important
    &:hover
      color: var(--color-white) !important
    @media (min-width: $middle-min)
      margin-bottom: .5rem
    @media (min-width: $extra-small-min)
      br
        display: none

  &__social
    @media (max-width: $middle-max)
      margin: 1rem 0
