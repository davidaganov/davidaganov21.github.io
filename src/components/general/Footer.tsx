import Social from "../base/Social"

export default function Footer() {
  return (
    <footer
      className="footer"
      id="footer"
    >
      <div className="footer__inner inner">
        <a
          href="https://github.com/davidaganov21/davidaganov21.github.io"
          className="footer__author inline-link inline-link--white"
        >
          Developed & Designed&nbsp;
          <br />
          by David Aganov
        </a>
        <Social classes="footer__social" />
        <a
          className="footer__email inline-link inline-link--white"
          href="mailto:davidaganov21@gmail.com"
        >
          davidaganov21@gmail.com
        </a>
      </div>
    </footer>
  )
}
