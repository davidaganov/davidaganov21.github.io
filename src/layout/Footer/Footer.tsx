import cn from "classnames"
import styles from "./Footer.module.sass"
import { Social } from "../../components/"

export const Footer = (): JSX.Element => {
  return (
    <footer
      className={styles.footer}
      id="footer"
      role="contentinfo"
    >
      <div className={cn(styles.inner, "inner")}>
        <a
          href="https://github.com/davidaganov21/davidaganov21.github.io"
          className={cn(styles.author, "inline-link inline-link--white")}
        >
          Developed & Designed&nbsp;
          <br />
          by David Aganov
        </a>
        <Social className={styles.social} />
        <a
          className="inline-link inline-link--white"
          href="mailto:davidaganov21@gmail.com"
        >
          davidaganov21@gmail.com
        </a>
      </div>
    </footer>
  )
}
