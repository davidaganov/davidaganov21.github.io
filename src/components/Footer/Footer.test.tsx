import { render, screen } from "@testing-library/react"
import { Footer } from "./Footer"

describe("Footer component", () => {
  it("Footer renders", () => {
    render(<Footer />)

    expect(screen.getByText(/davidaganov21@gmail.com/i)).toBeInTheDocument()
    expect(screen.getByText(/developed & designed/i)).toBeInTheDocument()
  })

  it("Footer snapshot", () => {
    const title = render(<Footer />)

    expect(title).toMatchSnapshot()
  })
})
