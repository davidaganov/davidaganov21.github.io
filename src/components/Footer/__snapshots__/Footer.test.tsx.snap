// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component Footer snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <footer
        class="footer"
        id="footer"
        role="contentinfo"
      >
        <div
          class="inner inner"
        >
          <a
            class="author inline-link inline-link--white"
            href="https://github.com/davidaganov/davidaganov.github.io"
            rel="noreferrer"
            target="_blank"
          >
            Developed & Designed
             
            <br />
            by David Aganov
          </a>
          <ul
            class="social social"
          >
            <li
              class="item"
            >
              <a
                aria-label="Go to Github profile"
                class="link"
                href="https://github.com/davidaganov"
                rel="noreferrer"
                target="_blank"
              >
                <svg>
                  github.svg
                </svg>
              </a>
            </li>
            <li
              class="item"
            >
              <a
                aria-label="Go to Linkedin profile"
                class="link"
                href="https://www.linkedin.com/in/david-aganov/"
                rel="noreferrer"
                target="_blank"
              >
                <svg>
                  linkedin.svg
                </svg>
              </a>
            </li>
          </ul>
          <a
            class="inline-link inline-link--white"
            href="mailto:davidaganov21@gmail.com"
          >
            davidaganov21@gmail.com
          </a>
        </div>
      </footer>
    </div>
  </body>,
  "container": <div>
    <footer
      class="footer"
      id="footer"
      role="contentinfo"
    >
      <div
        class="inner inner"
      >
        <a
          class="author inline-link inline-link--white"
          href="https://github.com/davidaganov/davidaganov.github.io"
          rel="noreferrer"
          target="_blank"
        >
          Developed & Designed
           
          <br />
          by David Aganov
        </a>
        <ul
          class="social social"
        >
          <li
            class="item"
          >
            <a
              aria-label="Go to Github profile"
              class="link"
              href="https://github.com/davidaganov"
              rel="noreferrer"
              target="_blank"
            >
              <svg>
                github.svg
              </svg>
            </a>
          </li>
          <li
            class="item"
          >
            <a
              aria-label="Go to Linkedin profile"
              class="link"
              href="https://www.linkedin.com/in/david-aganov/"
              rel="noreferrer"
              target="_blank"
            >
              <svg>
                linkedin.svg
              </svg>
            </a>
          </li>
        </ul>
        <a
          class="inline-link inline-link--white"
          href="mailto:davidaganov21@gmail.com"
        >
          davidaganov21@gmail.com
        </a>
      </div>
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
